### DISCUSSION TITLE
How to create custom link dialog

### DISCUSSION THREAD

**Discussion by mimpushkarev** [mdx-editor/editor#206] [Q&A] [answered]
In case i need custom link dialog, for better UX with link suggestions. Can i override default dialog component or i need to recreate `linkDialogPlugin`, what the better solution?

**Comment by petyosi** [comment]
I would recommend that you look into what you want to override and see what's missing as a configuration option for the link dialog plugin (e.g. a custom component override of some sort). Happy to accept a PR in that direction.

**Reply by mimpushkarev** [reply, marked as answer]
I was wondering if u could check it
PR: https://github.com/mdx-editor/editor/pull/209

**Reply by petyosi** [reply]
Merged that. I was thinking of something more granular, but this works of course. 

If you have the time, I would appreciate if you can share an example code snippet - that would help other users who need such behavior.

**Reply by mimpushkarev** [reply]
The Mdx Editor has a possibility to customize any component except the Link Dialog. In my opinion, I need to create a component that looks similar to my UI.

About this solution, it can help:
- Create better user experiences using other UI libraries (like AntDesign)
- Get more flexibility, at showing hints that can help users
